app/core/store/modules/layoutChrome.js

```diff
@@ -0,0 +1,56 @@
+export default {
+  namespaced: true,
+
+  state () {
+    // TODO: Currently saving volume to session instead of database.
+    let cachedSound
+    if (window.sessionStorage) {
+      cachedSound = window.sessionStorage.getItem('layoutChrome/soundOn')
+    }
+
+    return {
+      soundOn: cachedSound !== 'false',
+      currentCourseInstanceId: null,
```

Comment:
Can this be read from the campaign module directly instead of persisted here?  The campaign store module already has the data and duplicating it means we need to track it in two locations now so we no longer have a source of truth.

Reply:
It seems like the campaign and course instance id are used to determine the link in the layout component below.  That logic could be moved to a getter here `getLinkUrl` or could be reduced to `isPlayingInClassroom`.  The component could then use this method to determine what link to show.  It would still need to `mapGetters` for the campaign Id and course instance ID in this case.

Reply:
The two important points are to keep the state clean (one source of truth) and to encapsulate the business logic as state where possible.  Doing this makes the code much easier to follow and much easier to change in the future.

Reply:
I will reduce this to `getMapUrl` so that we can iterate o the state without coupling the layoutChrome to  the state of the module. Great points above.

Reply:
By reducing this into a business logic getter, I was also able to add a test which will ensure this doesn't change.